src/widget/woverview.cpp

```diff
@@ -400,11 +401,10 @@ void WOverview::onPassthroughChange(double v) {
     update();
 }
 
-void WOverview::slotTypeChanged(double v) {
-    int type = static_cast<int>(v);
-    VERIFY_OR_DEBUG_ASSERT(type >= 0 && type <= 2) {
-        type = 2;
-    }
+void WOverview::slotTypeControlChanged(double v) {
+    // Assert that v is in enum range to prevent UB.
+    DEBUG_ASSERT(v >= 0 && v < QMetaEnum::fromType<Type>().keyCount());
+    Type type = static_cast<Type>(static_cast<int>(v));
```

Comment:
is the double cast really required? A single `static_cast<Type>(v)` seemed sufficient from my testing.

Reply:
yes, the simple cast works, just did that out of anticipatory obedience to clang-tidy. will revert it.

Reply:
don't think clang tidy will complain, its tuned with very few rules. I don't even there is a rule (nor a justification) for requiring two casts here.

Reply:
I just found this note, not sure if it still applies:
https://github.com/mixxxdj/mixxx/blob/796d056a584b6667278355d8a3ba840e9198e78e/src/engine/controls/cuecontrol.cpp#L2734-L2736

Reply:
just checked, it does (compiler explorer with msvc v19.0 (Wine)), even though [it shouldn't](https://developercommunity.visualstudio.com/t/the-compiler-emits-c2440-when-trying-to-static-cas/793475).

Reply:
Sorry for the bogus comment then.